Doc/c-api/long.rst

```diff
@@ -85,13 +85,12 @@ All integers are implemented as "long" integer objects of arbitrary size.
    Return a new :c:type:`PyLongObject` based on the string value in *str*, which
    is interpreted according to the radix in *base*.  If *pend* is non-*NULL*,
    *\*pend* will point to the first character in *str* which follows the
-   representation of the number.  If *base* is ``0``, the radix will be
-   determined based on the leading characters of *str*: if *str* starts with
-   ``'0x'`` or ``'0X'``, radix 16 will be used; if *str* starts with ``'0o'`` or
-   ``'0O'``, radix 8 will be used; if *str* starts with ``'0b'`` or ``'0B'``,
-   radix 2 will be used; otherwise radix 10 will be used.  If *base* is not
-   ``0``, it must be between ``2`` and ``36``, inclusive.  Leading spaces are
-   ignored.  If there are no digits, :exc:`ValueError` will be raised.
+   representation of the number.  If *base* is ``0``, the *str* is interpreted using
+   the :ref:`integers` defintion; additionally, a literal 0 evaluates as *base* 0 and
+   a :attr:`decinteger` beginning with 0 raises a :exc:`ValueError`.  If *base* is not
```

Comment:
The rest of the text refers to the first argument by name as `*str*`, not “the `*str*`”, so I would drop “the” and just write “If `*base*` is 0, `*str*` is interpreted . . .”

Spelling: defintion → definition.

What does “a literal 0 evaluates as `*base*` 0” mean? This is already qualified with “if `*base*` is 0”.

I understand the “:attr:” markup is supposed to be for class attributes. Is “decinteger” a reference to the grammar definition:

decinteger: nonzerodigit (["`_`"] digit)* | "0"+ (["`_`"] "0")*

You may have to add more text to make this obvious, or explain what you mean by one “beginning with 0”.